### PR TITLE
Create an EKS cluster ready to use component

### PR DESCRIPTION
**Pull request by KevinFairise2**
What does this PR do?
---------------------

Create a component for creating an EKS cluster. With a simple API to begin with

Which scenarios this will impact?
-------------------

Motivation
----------

Additional Notes
----------------

### CODE REVIEW

File: scenarios/aws/eks/cluster.go [hunk 1]
```diff
@@ -0,0 +1,279 @@
+package eks
+
+import (
+	"github.com/DataDog/test-infra-definitions/common/config"
+	"github.com/DataDog/test-infra-definitions/common/utils"
+	"github.com/DataDog/test-infra-definitions/components"
+	kubecomp "github.com/DataDog/test-infra-definitions/components/kubernetes"
+	"github.com/DataDog/test-infra-definitions/resources/aws"
+	localEks "github.com/DataDog/test-infra-definitions/resources/aws/eks"
```
Comment (pducolin) [inline scenarios/aws/eks/cluster.go:9]: **🥜 nitpick**
Would rename this to `awseks`, at first I thought it was referencing local kind

File: tasks/aws/eks.py [hunk 2]
```diff
@@ -90,7 +90,7 @@ def create_eks(
 
 def _show_connection_message(ctx: Context, full_stack_name: str, config_path: Optional[str]):
     outputs = tool.get_stack_json_outputs(ctx, full_stack_name)
-    kubeconfig_output = json.loads(outputs["dd-Cluster-aws-eks"]["kubeConfig"])
+    kubeconfig_output = json.loads(outputs["dd-Cluster-eks"]["kubeConfig"])
```
Comment (pducolin) [inline tasks/aws/eks.py:93]: **👏 praise**

**Review by pducolin — APPROVED** [review]
Approving, a double check by the containers team would be better

File: scenarios/aws/eks/cluster.go (hunk 1 above)
Comment (KevinFairise2) [reply scenarios/aws/eks/cluster.go:9]: Agree, I just copy pasted what existed, but that looks better to change it to `awseks`

Comment (KevinFairise2) [reply scenarios/aws/eks/cluster.go:9]: Well actually we already have another package named `awseks` refering to `github.com/pulumi/pulumi-aws/sdk/v6/go/aws/eks` . Here `localeks` refer to resource defined in the local repository (`test-infra-definitions`)

**Review by vboulineau — APPROVED** [review]

**Pull request merged**